### ISSUE TITLE
[HOLD for payment 2023-08-01] [$250] Migrate index.js to function component

### ISSUE THREAD

**Issue by marcaaron** [Expensify/App#16183]
### Filenames
- [ ] [src/components/OptionsList/index.js](https://github.com/Expensify/App/blob/main/src/components/OptionsList/index.js)
    - **type:** `React.Component`
    - **has state values:** false
    - **has refs:** true
    - **has context:** false
    - **lifecycle methods:** `componentDidMount`,`componentWillUnmount`

### Task
- We currently have some class components in our codebase that we would like to refactor to a function component.
- Here's a link with some general advice on how to refactor a class component to a function component: https://react.dev/reference/react/Component#alternatives
- If you need additional guidance, please ask in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q)
- Test for any regressions and verify that there are no breaking changes

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b30e5df5faab3be2</li>
        <li>Upwork Job ID: 1681755244114116608</li>
        <li>Last Price Increase: 2023-07-19</li>
    </ul>
</details>

**Comment by s-alves10** [comment]
I'd love to work on this.

**Comment by dayana7204** [comment]
I am ready to work on this ticket. :)

**Comment by olexyt** [comment]
I can work on this issue.

**Comment by mejed-alkoutaini** [comment]
I'm able to work on this if needed!

**Comment by mkhutornyi** [comment]
I'd like to work on this.

**Comment by kosmydel** [comment]
Hey, I'm from the SWM expert's agency. I would like to work on this.

**Comment by alexxxwork** [comment]
I would love to work on this issue

**Comment by vadymbokatov** [comment]
I can work on this issue

**Comment by reinesarchisorte** [comment]
Could i take this one?

**Comment by reinesarchisorte** [comment]
Contributor details
Your Expensify account email: lev.dvor1@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~010a93f82c510e25dd

**Comment by legend524** [comment]
Contributor details
Your Expensify account email: thecrazydev524@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01ec76419a7c2c94b0

**Comment by legend524** [comment]
[Proposal]
Here are my plans to solve this issue.
1. in component definition, upgrade this code
class OptionsList extends Component { 
to
 const OptionsList = (props, ref) => {

1. Upgrade state definition using useState
        this.touchStart = this.touchStart.bind(this);
        this.touchEnd = this.touchEnd.bind(this);

3. Upgrade componentDidMount and componentWillUnmount using useEffect

4. remove all of these code.
OptionsList.propTypes = {
    ...propTypes,
};
OptionsList.defaultProps = defaultProps;

export default withWindowDimensions(
    forwardRef((props, ref) => (
        <OptionsList
            forwardedRef={ref}
            // eslint-disable-next-line react/jsx-props-no-spreading
            {...props}
        />
    )),
);

and simply replace with this code.

export default forwardRef(OptionsList);

4. some more refactoring...

**Comment by alitoshmatov** [comment]
Can I work on this one

**Comment by kevinksullivan** [comment]
Paying out next week

**Comment by marcaaron** [comment]
Not overdue just waiting for payment i believe

**Comment by JmillsExpensify** [comment]
@kevinksullivan can you summarize the proper payouts on this issue?

**Comment by kevinksullivan** [comment]
Payment summary:
- C+: @parasharrajat - $250
